src/scan-imports.ts

```diff
@@ -63,34 +64,24 @@ function parseWebModuleSpecifier(specifier: string, knownDependencies: string[])
   if (webModulesIndex === -1) {
     return null;
   }
-  // check if the resolved specifier (including file extension) is a known package.
+
+  // check if the resolved specifier without extension matches npm name format
   const resolvedSpecifier = cleanedSpecifier.substring(webModulesIndex + WEB_MODULES_TOKEN_LENGTH);
-  if (knownDependencies.includes(resolvedSpecifier)) {
-    return resolvedSpecifier;
-  }
-  // check if the resolved specifier (without extension) is a known package.
   const resolvedSpecifierWithoutExtension = stripJsExtension(resolvedSpecifier);
-  if (knownDependencies.includes(resolvedSpecifierWithoutExtension)) {
+  if (validate(resolvedSpecifierWithoutExtension).validForNewPackages) {
     return resolvedSpecifierWithoutExtension;
   }
   // Otherwise, this is an explicit import to a file within a package.
   return resolvedSpecifier;
 }
 
-function getInstallTargetsForFile(
-  filePath: string,
-  code: string,
-  knownDependencies: string[],
-): InstallTarget[] {
+function getInstallTargetsForFile(filePath: string, code: string): InstallTarget[] {
   const allImports: InstallTarget[] = [];
   try {
     const ast = parse(code, {plugins: ['dynamicImport'], sourceType: 'module'});
     traverse(ast, {
       ImportDeclaration(path) {
-        const webModuleSpecifier = parseWebModuleSpecifier(
-          path.node.source.value,
-          knownDependencies,
-        );
+        const webModuleSpecifier = parseWebModuleSpecifier(path.node.source.value);
```

Comment:
woo! Excited to see us removing that concept of knownDependencies and replacing it with something less stateful / totally static parsing!